.github/workflows/build-Dockerfile

```diff
@@ -29,8 +29,23 @@ LABEL version=$PRODUCT_VERSION
 # Historical Terraform-specific label preserved for backward compatibility.
 LABEL "com.hashicorp.terraform.version"="${PRODUCT_VERSION}"
 
```

Comment:
Question for RelEng: Is this the actual Dockerfile that is in use today? Because there is also https://github.com/hashicorp/terraform-releases/blob/master/build-support/docker-release/Dockerfile-release but that has been updated a lot less recently (last update was 4 years ago) so my assumption is that is no longer used.

Reply:
Yes afaik the one in terraform-releases is a legacy one from before y'all onboarded to our Common Release Tooling - you have the correct one here (`build-Dockerfile`)